### ISSUE TITLE
Building app with correct environment variables

### ISSUE THREAD

**Issue by Sanjo** [thereactivestack/meteor-webpack#49]
Hi, yesterday I found a problem where our app got deployed with the development environment variables instead of the production environment variables.

To fix the issue I had to:

```
# Set app environment variables here also
meteor reset
mupx deploy
```

So building the app with the correct environment variables didn't happen without `meteor reset`.

**Comment by Sanjo** [comment]
After investigating again, the problem was only that I didn't set the new environment variables before calling `mupx deploy`. Doing that fixes the problem. No need for `meteor reset`.

**Comment by eXon** [comment]
We'll stop supporting NODE_ENV variable and just check if Meteor is built in production or not so I'm not sure this is going to be a problem.

**Comment by Sanjo** [comment]
The issue was with other environment variables that I use in my app. With Mupx you usually configure them in the mup.json, but that doesn't work currently with webpack:webpack. Instead you need to set the environment variables in your deploy environment.

If you remove the feature that replaces environment variables in code with constants, this is no longer an issue.

**Comment by eXon** [comment]
Yea this is because webpack optimize the process.env. I'm looking for a way to fix that. For now, you can also use a meteor.startup.js file and set it on a global variable.

**Comment by Sanjo** [comment]
Another alternative is to use Meteor settings.

**Comment by eXon** [comment]
This should be fixed on the 1.0 release because we target node. This means the environment variable set on the server will be available :)
